src/io/fld-read.jl

```diff
@@ -247,65 +247,29 @@ end
 """
 `format, endian, bytes = datatype_fld_to_mat(datatype)`
 
-determine data format from .fld header datatype
+Determine data format from .fld header datatype.
 """
 function datatype_fld_to_mat(datatype::AbstractString)
-	if datatype == "byte"
-		format = UInt8
-		endian = "ieee-be" # irrelevant
-		bytes = 1
-
-	elseif datatype in ["short_be", "short_sun", "xdr_short"]
-		format = Int16
-		endian = "ieee-be"
-		bytes = 2
-	elseif datatype == "short_le"
-		format = Int16
-		endian = "ieee-le"
-		bytes = 2
-
-	elseif datatype == "int"
-		format = Int32
-		endian = "" # native int - not portable
-		bytes = 4
-	elseif datatype == "int_le"
-		format = Int32
-		endian = "ieee-le"
-		bytes = 4
-	elseif datatype in ["int_be", "xdr_int"]
-		format = Int32
-		endian = "ieee-be"
-		bytes = 4
-
-	elseif datatype == "float"
-		format = Float32 # typeof(1.)
-		endian = "" # native float - not portable
-		bytes = 4
-	elseif datatype == "float_le"
-		format = Float32
-		endian = "ieee-le"
-		bytes = 4
-	elseif datatype in ["float_be", "xdr_float"]
-		format = Float32
-		endian = "ieee-be"
-		bytes = 4
-
-	elseif datatype == "double"
-		format = Float64
-		endian = "" # native double - not portable
-		bytes = 8
-	elseif datatype == "double_le"
-		format = Float64
-		endian = "ieee-le"
-		bytes = 8
-	elseif datatype in ["double_be", "xdr_double"]
-		format = Float64
-		endian = "ieee-be"
-		bytes = 8
-
-	else
-		throw("format '$datatype' not yet implemented. ask jeff!")
-	end
 
-	return format, endian, bytes
+	dict = Dict([
```

Comment:
Compared to the current implementation, Using `Dict` might be bad in performance.

```julia
# after
julia> @btime datatype_fld_to_mat("byte")
  1.822 μs (60 allocations: 3.81 KiB)
(UInt8, :be, 1)

# Before
julia> @btime datatype_fld_to_mat("byte")
  50.033 ns (1 allocation: 32 bytes)
(UInt8, "ieee-be", 1)
```

Reply:
Thanks very much for the info!  Then I will avoid using `Dict` in performance critical places.  Here it is just some IO for an obscure file type used rarely so I think this version is easier to read.
I'm mostly working to improve code coverage at the moment...

Reply:
BTW, is there a `const` variant that would be better to use?  I couldn't find one...

Reply:
> BTW, is there a const variant that would be better to use? I couldn't find one...

I don't know much about it, but I guess `NamedTuple` might work

Reply:
Thanks - good idea.  Unfortunately the keys have `-` in the string so it would be some work.  Maybe later...